src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
Agree. I would check the complete error message. Although it is not the first time we disscuss this in a PR and I think that @evanrolfe argument was that if you modify the error message because then you don't have to change the test.